src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/attachment/v2/AwsV2AttachmentPlugin.java

```diff
@@ -0,0 +1,174 @@
+import software.amazon.awssdk.services.ec2.model.Volume;
+import software.amazon.awssdk.services.ec2.model.VolumeAttachment;
+
+public class AwsV2AttachmentPlugin implements AttachmentPlugin<AwsV2User>{
+
+	private static final Logger LOGGER = Logger.getLogger(AwsV2VolumePlugin.class);
+	private static final String CLEAR_ATTACHMENT_ID = "";
+	private static final String RESOURCE_NAME = "Attachment";
+	private static final int FIRST_POSITION = 0;
+	private static final int SECOND_POSITION = 1;
+
+	protected static final String ATTACHMENT_ID_PREFIX = "att-";
+	protected static final String ATTACHMENT_ID_TAG = "attachment-id";
+	protected static final String DEFAULT_DEVICE_NAME = "/dev/sdh";
+	protected static final String FILTER_BY_TAG_ATTACHMENT_ID = "tag:attachment-id";
+	protected static final String XVDH_DEVICE_NAME = "xvdh";
+	
+	private String region;
+	
+	public AwsV2AttachmentPlugin(String confFilePath) {
+		Properties properties = PropertiesUtil.readProperties(confFilePath);
+		this.region = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_REGION_SELECTION_KEY);
+	}
+	
+	@Override
+	public boolean isReady(String instanceState) {
+		return AwsV2StateMapper.map(ResourceType.ATTACHMENT, instanceState).equals(InstanceState.READY);
+	}
+
+	@Override
+	public boolean hasFailed(String instanceState) {
+		return AwsV2StateMapper.map(ResourceType.ATTACHMENT, instanceState).equals(InstanceState.FAILED);
+	}
+
+	@Override
+	public String requestInstance(AttachmentOrder attachmentOrder, AwsV2User cloudUser) throws FogbowException {
+		LOGGER.info(String.format(Messages.Info.REQUESTING_INSTANCE, cloudUser.getToken()));
+
+		String device = defineDeviceNameAttached(attachmentOrder.getDevice());
+		String instanceId = attachmentOrder.getComputeId();
+		String volumeId = attachmentOrder.getVolumeId();
+		AttachVolumeRequest attachmentRequest = AttachVolumeRequest.builder()
+				.device(device)
+				.instanceId(instanceId)
+				.volumeId(volumeId)
+				.build();
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		client.attachVolume(attachmentRequest);
+		
+		String attachmentId = ATTACHMENT_ID_PREFIX + getRandomUUID();
+		CreateTagsRequest tagRequest = createTagAttachmentId(attachmentId, volumeId);
+		client.createTags(tagRequest);
+
+		return attachmentId;
+	}
+
+	@Override
+	public void deleteInstance(AttachmentOrder attachmentOrder, AwsV2User cloudUser) throws FogbowException {
+		LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE, attachmentOrder.getInstanceId(), cloudUser.getToken()));
+
+		String volumeId = attachmentOrder.getVolumeId();
+		DetachVolumeRequest detachVolumeRequest = DetachVolumeRequest.builder()
+				.volumeId(volumeId)
+				.build();
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		try {
+			client.detachVolume(detachVolumeRequest);
+			CreateTagsRequest tagRequest = createTagAttachmentId(CLEAR_ATTACHMENT_ID, volumeId);
```

Comment:
Why u create a tag after deleting the attach?

Reply:
Because the attachment is a resource of volume in which we created the tag "attachment-id" to place its ID when the detachment occurs the value of this tag must be removed from the volume that had been attached.